# This is a workaround to the crashes seen on Ubuntu.  Setting info to zero
# makes info.info.x11.display zero and avoids the calls to
# opengl_exec_set_parent_window, one of which is crashing.

Index: qemu-0.12.4/sdl.c
===================================================================
--- qemu-0.12.4.orig/sdl.c
+++ qemu-0.12.4/sdl.c
@@ -871,6 +871,7 @@ void sdl_display_init(DisplayState *ds, 
     vi = SDL_GetVideoInfo();
     host_format = *(vi->vfmt);
 
+    bzero(&info, sizeof(info));
     SDL_GetWMInfo(&info);
     if (info.subsystem == SDL_SYSWM_X11 && info.info.x11.display)
         opengl_exec_set_parent_window(info.info.x11.display,
